File: LinkedListCycle.js
class Node
{
    constructor(value)
    {
        this.value=value
        this.next=null
    }
}

function cycleDetection(head)
{
    let obj={}
    let temp=head 
    while(temp!=null)
    {
        if(temp in obj)
        {
            console.log(obj)
            return temp
        }
        else{
            console.log(obj)
            obj[temp.value]=true
        }
        temp=temp.next
    }
    return null
}

let node=new Node(1)
node.next=new Node(2)
node.next.next=new Node(3)
node.next.next.next=new Node(4)
node.next.next.next.next=new Node(5)
node.next.next.next.next.next=new Node(6)
node.next.next.next.next.next.next=new Node(7)
node.next.next.next.next.next.next.next=new Node(8)
node.next.next.next.next.next.next.next.next=new Node(9)
//node.next.next.next.next.next.next.next.next.next=node.next.next
console.log(cycleDetection(node))

File: LinkedListCycle.py
#Hash map 
#Linked List Cycle detection using brut force approach

class Node:
    def __init__(self,value):
        self.value=value
        self.next=None 
def cycleDetection(head):
    dict1={}
    temp=head 
    #trvaersing through linked list
    while temp:
        #checking the address
        if temp in dict1:
            return temp 
        else:
            #storing the linked list address if not found
            dict1[temp]=True
        #changing the address 
        temp=temp.next
def main():
    node=Node(1)
    node.next=Node(2)
    node.next.next=Node(3)
    node.next.next.next=Node(4)
    node.next.next.next.next=Node(5)
    node.next.next.next.next.next=Node(6)
    node.next.next.next.next.next.next=Node(7)
    node.next.next.next.next.next.next.next=Node(8)
    node.next.next.next.next.next.next.next.next=Node(9)
    node.next.next.next.next.next.next.next.next.next=node.next.next
    print(cycleDetection(node).value)
main()
    
    
           